service/awsconfig/drainer_framework.go

```diff
@@ -110,7 +110,9 @@ func NewDrainerFramework(config DrainerFrameworkConfig) (*framework.Framework, e
 			Logger:         config.Logger,
 			ResourceRouter: resourceRouter,
 
-			Name: config.ProjectName,
+			// Name is used to compute finalizer names. This here results in something
+			// like operatorkit.giantswarm.io/aws-operator-drainer.
+			Name: config.ProjectName + "-drainer",
```

Comment:
So this here just fixes one framework because we do not have a mechanism yet to removed unwanted finalizers. Thus this is a quick fix and should keep the deletion on AWS consistent. We can discuss in the original issue how to fix this in general.

Reply:
Approving to fix the current issue. Thanks for the comment. An explicit field for the finalizer name in the framework config would be good IMO.

Reply:
I thought about renaming `Name` as well. Will do later once we agree on some sustainable solution.